src/libraries/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/AnyOS/ManagedPal.KeyTrans.cs

```diff
@@ -168,7 +168,7 @@ internal ManagedKeyTransPal(KeyTransRecipientInfoAsn asn)
                 throw new CryptographicException(SR.Cryptography_Cms_UnknownAlgorithm);
             }
 
-            using (RSA rsa = recipient.Certificate.GetRSAPublicKey())
+            using (RSA rsa = recipient.Certificate.GetRSAPublicKey()!)
```

Comment:
In what situations will GetRSAPublicKey return null, and what guarantees those aren't the case here?

Reply:
> In what situations will GetRSAPublicKey return null,

When the cert key algorithm isn't RSA.

> and what guarantees those aren't the case here?

Way higher level than this, KTRI is only chosen if the key was RSA-typed.  Note all the hard-coded RSA-ness to the method.

Reply:
Great, thanks.